src/base/utils/string.h

```diff
@@ -71,5 +72,22 @@ namespace Utils
         TriStateBool parseTriStateBool(const QString &string);
 
         QString join(const QVector<QStringRef> &strings, const QString &separator);
+
+        template <typename U, typename std::enable_if<std::is_enum<U>::value, int>::type = 0>
+        QString serialize(const U &value)
```

Comment:
The intention was, on the contrary, to make them faceless, so as to have serialize/parse for different types afterwards. Do you think this is a bad idea?

Reply:
>Do you think this is a bad idea?

I'm not sure, but it is worth a try.

Also `serialize()` looks like `toString()` (or `toQString()`) to me.

Reply:
>Also `serialize()` looks like `toString()` (or `toQString()`) to me.

Then `parse()` looks like `fromString()`?

I can change them if you want.

Reply:
>I can change them if you want.

Yes, please.
Also a wild idea: maybe place them under `Utils::Enum` namespace too?

Reply:
>For this function to work, the enum type must be first declared with `Q_ENUM`. 

Unfortunately we can't statically assert on it (at least without using some Qt internals). So I can either add runtime assertion or make it to serialize regular enums (just use string representation of numbers).

Reply:
> >I can change them if you want.
> 
> Yes, please.
> Also a wild idea: maybe place them under `Utils::Enum` namespace too?

I want to summarize. Do you prefer to have multiple namespaces `Utils::XXX` with a pair of `fromString()`/`toString()` functions in each instead of a pair of `serialize()`/`parse()` templates in a single `Utils::String` namespace? Or a set of pairs of `fromXXX()`/`toXXX()` functions in a single `Utils::String` namespace?

P.S. `Utils::String::toString()` and `Utils::String::fromString()` looks weird/confusing for me.

Reply:
>Unfortunately we can't statically assert on it (at least without using some Qt internals). So I can either add runtime assertion or make it to serialize regular enums (just use string representation of numbers).

I haven't tried but maybe `QMetaEnum::fromType()` will emit error/warning when it is used with non-Q_ENUM types?
Otherwise we can add a runtime assertion (Q_ASSERT) for it.

>I want to summarize...

Never mind, the current `fromEnum`, `toEnum` is good.

Reply:
>I haven't tried but maybe `QMetaEnum::fromType()` will emit error/warning when it is used with `non-Q_ENUM` types?
Otherwise we can add a runtime assertion for it.

Well, it turns out that there is a **static assertion**, so nothing else is needed.